### PR TITLE
Remove sklearn dependency.

### LINKED ISSUE #590: Remove scikit-learn dependency
scikit-learn is only used for PCA to align molecular axes in SmilesWidget. Pulling such a heavy dependency just for this is an overkill. I am not sure if it is even still needed, and I am sure there must be less heavy-weight to do it.

https://stats.stackexchange.com/questions/235882/pca-in-numpy-and-sklearn-produces-different-results

Past PRs for context:
https://github.com/aiidalab/aiidalab-widgets-base/pull/160
https://github.com/aiidalab/aiidalab-widgets-base/pull/164

### PR DESCRIPTION
**Pull request by yakutovicha**
fixes #590 

Sklearn was only used to align SMILES-generated molecules with the Z-axis. The usefulness of such functionality is questionable, and it doesn't justify depending on such a heavy library.

### CODE REVIEW

**Comment by yakutovicha** [comment]
We've also discussed this with @cpignedoli (who originally introduced that functionality) and he agrees.

**Review by danielhollas — CHANGES_REQUESTED** [review]
Looks good, thanks so much for doing this!

I think we should have a unit test that is sensitive the the generated coordinates from the smiles code. Could you maybe extend the existing tests/test_structure.py::test_smiles_widget() to assert the coordinates of some of the molecules? (perhaps the N2 is simplest). I would use the `np.allclose()` of comparisons to reference so that we're not sensitive to minor changes in the rdkit library.

**Comment by danielhollas** [comment]
The Firefox UI tests failures are related to #672, no need to worry about that here.

**Comment by yakutovicha** [comment]
> I think we should have a unit test that is sensitive the the generated coordinates from the smiles code.

Hey @danielhollas, I've been thinking about this. Are we sure this is needed?

Suppose the structure that comes out of rdkit is rotated - is it a problem? The tests will fail, but the only thing we will do to fix them is to change the tests.

It is not a problem to add such a check, I don't necessary see the benefits of doing that.

**Comment by danielhollas** [comment]
Yeah, I don't feel strongly about it. It just feels weird that the change in this PR did not break any tests. Feel free to merge, we can revisit later.

**Review by danielhollas — APPROVED** [review]
